docs/decisions/0011_multiple_event_types_per_topic.rst

```diff
@@ -0,0 +1,74 @@
+10. Multiple event types per topic
+##################################
+
+Status
+******
+
+- **Accepted** DATE TO BE ADDED BEFORE MERGING
+
+Context
+*******
+
+For the initial implementation of the event bus, we decided to limit each topic to use a single schema. This meant that every signal in openedx-events required a different topic. This worked for our initial use case, course catalog updates, because all changes were considered updates and could be emitted using the same signal.
+However, other types of events are not so easily grouped into a single signal. For example, there are different signals for ``XBLOCK_PUBLISHED``, ``XBLOCK_UPDATED``, and ``XBLOCK_DUPLICATED``, and indeed different schemas across the three signals. Routing these signals through different topics could lead to events being processed in a nonsensical order, for example an xblock being duplicated before it's been published. See `Should You Put Several Event Types in the Same Kafka Topic?`_ and `Putting Several Event Types in the Same Topic – Revisited`_ for more information on why it can be useful to group different event types on the same topic.
+
+.. _Should You Put Several Event Types in the Same Kafka Topic?: https://www.confluent.io/blog/put-several-event-types-kafka-topic/
+.. _Putting Several Event Types in the Same Topic – Revisited: https://www.confluent.io/blog/multiple-event-types-in-the-same-kafka-topic/
+
+Decision
+********
+We will update the code for event serializers, which are responsible for registering and validating schemas against the schema registry, to use the TopicRecordName strategy.
```

Comment:
Maybe start with something like:

> We will allow multiple related event types to be produced to a single topic so that they can be ordered appropriately. We will accomplish this by...

Reply:
[resolved]